src/OperationProcessor.ts

```diff
@@ -252,6 +257,15 @@ class OperationProcessorImpl implements OperationProcessor {
     // previous info if it exists
     this.opHashToInfo.set(opHash, opInfo);
 
+    if (operation.type === OperationType.Delete ||
+        operation.type === OperationType.Recover) {
+      // NOTE: only assuming and hanldling delete currently.
+      // TODO: validate recovery key.
+
+      this.deletedDids.add(operation.did!);
```

Comment:
Don't we have to verify the signature for delete operation? This requirement introduces a dependency between the delete operation and its ancestors (which might define the public key to be used for verification) - which means that if some ancestor is missing and therefore the delete operation is unvalidated, we might have to postpone adding the deleted dids to the deletedDids set.

Reply:
Sounds good - thanks.